spinalcordtoolbox/utils/sys.py

```diff
@@ -141,6 +141,9 @@ def set_loglevel(verbose):
     # Set the logging level globally, but only when scripts are directly invoked (e.g. from the command line)
     if caller_module_name == "__main__":
         logging.root.setLevel(getattr(logging, log_level))
+        # matplotlib is particularly chatty, so keep it at the default level
+        # see: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4086
+        logging.getLogger('matplotlib').setLevel(logging.INFO)
```

Comment:
Just so I'm understanding here: Is the core issue because:

-  _We_ set the root loglevel to `debug` (via `-v 2`), and thus accidentally increased the logging coming from matlplotlib? 
- Or, is it that Matplotlib's `findfont: score(` messages are `INFO` messages, meaning that they sneak through regardless of what `-v` is set to in a script.

If it's the latter, we may want to use `logging.WARNING` instead, But if it's the former, then this LGTM.

Reply:
Good suggestion, I'll change it to `logging.WARNING`.

The specific `findfont: score` messages are `logging.DEBUG` level, since they're produced [here](https://github.com/matplotlib/matplotlib/blob/b3bd929cf07ea35479fded8f739126ccc39edd6d/lib/matplotlib/font_manager.py#L1390). This is in the `matplotlib.font_manager` module, which [uses its `__name__` for its logger](https://github.com/matplotlib/matplotlib/blob/b3bd929cf07ea35479fded8f739126ccc39edd6d/lib/matplotlib/font_manager.py#L47).

My understanding is that in the normal situation, as per [the defaults](https://docs.python.org/3/library/logging.html#logging.Logger.setLevel) (that is, the root logger is at `logging.WARNING` and all other loggers are at `logging.NOTSET`):
1. The message gets created with a `logging.DEBUG` level in the `matplotlib.font_manager` logger.
2. The `matplotlib.font_manager` logger is at `logging.NOTSET`, so the message percolates up.
3. The `matplotlib` logger is at `logging.NOTSET`, so it percolates up again.
4. The root logger is at `logging.WARNING`, so the message gets discarded.

When we set the root logger to `logging.DEBUG`, that step 4 suddenly lets the message through to the console. If we want to get only the "default" logging with matplotlib, then setting `matplotlib` to `logging.WARNING` would make make the message get filtered out in step 3, correctly.